src/main/java/ast/CreateDDB.java

```
@@ -1,20 +1,107 @@
 package ast;
 
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.dynamodbv2.model.*;
+import libs.TableUtils;
+
+/**
+ * CREATE-DDB ::= "create-ddb {"
+ * "name:" STRING
+ * "region": STRING
+ * "key:" STRING
+ * "key-type": "string" | "number"
+ * "sort-key:" STRING
+ * "sort-key-type": "string" | "number"
+ * "read-capacity-units:" NUMBER
+ * "write-capacity-units:" NUMBER
+ * "}"
+ */
 public class CreateDDB extends Create {
     private String name;
+    private String region;
     private String key;
+    private String keyType;
     private String sortKey;
+    private String sortKeyType;
     private Long readCapacityUnits;
     private Long writeCapacityUnits;
 
     @Override
     public void parse() {
-        // todo
+        tokenizer.checkAndGetNext("create-ddb");
+        tokenizer.checkAndGetNext("\\{");
+
+        tokenizer.checkAndGetNext("name:");
+        name = tokenizer.getNext();
+
+        tokenizer.checkAndGetNext("region:");
+        region = tokenizer.getNext();
+
+        tokenizer.checkAndGetNext("key:");
+        key = tokenizer.getNext();
+
+        tokenizer.checkAndGetNext("key-type:");
+        keyType = tokenizer.getNext();
+
+        tokenizer.checkAndGetNext("sort-key:");
+        sortKey = tokenizer.getNext();
+
+        tokenizer.checkAndGetNext("sort-key-type:");
+        sortKeyType = tokenizer.getNext();
+
+        tokenizer.checkAndGetNext("read-capacity-units:");
+        readCapacityUnits = Long.parseLong(tokenizer.getNext());
```

Where should we propagate errors if they try to give this field a string?

---

Good question, maybe line 57 can be captured in a try-block and we can throw an exception similar if we catch a `NumberFormatException` while parsing, something like
`throw new Exception("Read capacity units must be an integer");`